# Description: set the load path for chef-solr's binaries correctly for Debian
# FHS compliant location. No plans for upstream to change this.
Index: chef-0.9.0/chef-solr/bin/chef-solr
===================================================================
--- chef-0.9.0.orig/chef-solr/bin/chef-solr	2010-06-28 17:51:19.000000000 -0600
+++ chef-0.9.0/chef-solr/bin/chef-solr	2010-06-28 17:51:21.000000000 -0600
@@ -19,8 +19,8 @@
 
 $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "lib")))
 $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "chef", "lib")))
+$:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "share", "chef-solr", "lib")))
 
-require 'rubygems'
 require 'chef/solr/application/solr'
 
 Chef::Solr::Application::Solr.new.run
Index: chef-0.9.0/chef-solr/bin/chef-solr-indexer
===================================================================
--- chef-0.9.0.orig/chef-solr/bin/chef-solr-indexer	2010-06-28 17:51:19.000000000 -0600
+++ chef-0.9.0/chef-solr/bin/chef-solr-indexer	2010-06-28 17:51:21.000000000 -0600
@@ -19,8 +19,8 @@
 
 $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "lib")))
 $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "chef", "lib")))
+$:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "share", "chef-solr", "lib")))
 
-require 'rubygems'
 require 'chef/solr/application/indexer'
 
 Chef::Solr::Application::Indexer.new.run
Index: chef-0.9.0/chef-solr/bin/chef-solr-rebuild
===================================================================
--- chef-0.9.0.orig/chef-solr/bin/chef-solr-rebuild	2010-06-28 17:51:19.000000000 -0600
+++ chef-0.9.0/chef-solr/bin/chef-solr-rebuild	2010-06-28 17:51:21.000000000 -0600
@@ -19,8 +19,8 @@
 
 $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "lib")))
 $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "chef", "lib")))
+$:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "share", "chef-solr", "lib")))
 
-require 'rubygems'
 require 'chef/solr/application/rebuild'
 
 Chef::Solr::Application::Rebuild.new.run
Index: chef-0.9.0/chef-server-api/lib/chef-server-api.rb
===================================================================
--- chef-0.9.0.orig/chef-server-api/lib/chef-server-api.rb	2010-06-28 17:47:50.000000000 -0600
+++ chef-0.9.0/chef-server-api/lib/chef-server-api.rb	2010-06-28 17:51:21.000000000 -0600
@@ -1,2 +1,3 @@
+$:.unshift File.join(File.dirname(__FILE__), "..", "..", "chef-solr", "lib")
 CHEF_SERVER_VERSION = "0.9.10.rc.2"
 CHEF_SERVER_API_ROOT = File.expand_path(File.dirname(__FILE__) + '/../')
Index: chef-0.9.0/chef-server-webui/lib/chef-server-webui.rb
===================================================================
--- chef-0.9.0.orig/chef-server-webui/lib/chef-server-webui.rb	2010-06-28 17:51:33.000000000 -0600
+++ chef-0.9.0/chef-server-webui/lib/chef-server-webui.rb	2010-06-28 17:52:01.000000000 -0600
@@ -1,2 +1,3 @@
+$:.unshift File.join(File.dirname(__FILE__), "..", "..", "chef-solr", "lib")
 CHEF_SERVER_WEBUI_VERSION = "0.9.10.rc.2"
 CHEF_SERVER_WEBUI_ROOT = File.expand_path(File.dirname(__FILE__) + '/../')
